src/Sulu/Bundle/CategoryBundle/Tests/Unit/Category/KeyWordManagerTest.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\CategoryBundle\Tests\Unit\Category;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Prophecy\Argument;
+use Sulu\Bundle\CategoryBundle\Category\KeyWordManager;
+use Sulu\Bundle\CategoryBundle\Category\KeyWordRepositoryInterface;
+use Sulu\Bundle\CategoryBundle\Entity\Category;
+use Sulu\Bundle\CategoryBundle\Entity\CategoryTranslation;
+use Sulu\Bundle\CategoryBundle\Entity\KeyWord;
+
+class KeyWordManagerTest extends \PHPUnit_Framework_TestCase
+{
+    public function provideSaveData()
+    {
+        return [
+            [],
+            [true],
+            [true, true],
+            [false, true],
+        ];
+    }
+
+    /**
+     * @dataProvider provideSaveData
+     */
+    public function testSave($exists = false, $has = false, $keyWordString = 'Test', $locale = 'de')
```

Comment:
I think I would not use default values here... That's quite confusing, since the value since the values in the dataprovider are quite hard to compare this way :confused: